### DISCUSSION TITLE
Limit max executing time for cron tasks

### DISCUSSION THREAD

**Discussion by fermion2020** [tiredofit/docker-nextcloud#50] [Q&A]
**Description of the feature**
At times, I've encountered situations where cron tasks became unresponsive, causing numerous processes to run and hang indefinitely, thus consuming significant resources.

I believe implementing a maximum execution time limit for cronjobs using the `timeout` command is a prudent idea. This measure should prevent any potential issues, provided the timeout duration is reasonable. For instance, setting a timeout of 30 minutes for tasks scheduled every 5 minutes should ensure smooth operation without risking system stability.

**Benftits of feature**

Having better control of background tasks.

**Additional context**

**Comment by tiredofit** [comment]
It may be beneficial to try looking into this nextcloud option to see if it would solve your issues: https://docs.nextcloud.com/server/28/admin_manual/configuration_server/background_jobs_configuration.html#parameters

I can build that into an environment variable for future releases, but something about killing a process thats long running repeatedly seems like it could eventually come back and hurt.

Regardless, this will likely end up as a feature in an upcoming base image (alpine/debian) release..
